library/cool/invariance.red

```diff
@@ -1,9 +1,126 @@
 import prelude
-import data.bool
+import data.list
+import data.nat
+import data.unit
+import paths.list
+import basics.isotoequiv
+
+def nat→list : nat → list unit =
+  elim [
+  | zero → nil
+  | suc (_ → xs) → cons triv xs
+  ]
+
+def nat→list/iso : iso nat (list unit) =
+  (nat→list,
+   length unit,
+   elim [
+   | nil → refl
+   | cons * (_ → ih) → λ i → cons triv (ih i)
+   ],
+   elim [
+   | zero → refl
+   | suc (_ → ih) → λ i → suc (ih i)
+   ])
+
+def nat→list/equiv : equiv^1 nat (list unit) = iso→equiv _ _ nat→list/iso
+def nat→list/path : path^1 type nat (list unit) = ua _ _ nat→list/equiv
+
+-- We can transport functions between these two implementations, and run them!
+--   from nat → nat to list unit → list unit...
+
+def double/nat : nat → nat =
+  elim [
+  | zero → zero
+  | suc (_ → ih) → suc (suc ih)
+  ]
+
+def double/list : list unit → list unit =
+  coe 0 1 double/nat in λ i → nat→list/path i → nat→list/path i
+
+def double/list/one : list unit = double/list (cons triv nil)
+meta ⦉ print normalize double/list/one ⦊
```

Comment:
btw it should be possible to inline this: `double/list (cons triv nil)` is synthesizable.

Reply:
Hm, I get a parse error; what is the syntax? Do I need some brackets?

Reply:
Yes, I think you might need parentheses around the expression part.